rhombus/rhombus/scribblings/ref-rx-object.scrbl

```diff
@@ -0,0 +1,241 @@
+ a start-of-file @rhombus(bof, ~at rhombus/rx) cannot.
+
+@examples(
+  ~eval: rx_eval
+  ~repl:
+    rx'^ "a"*'.match_in("aaa")
+    rx'^ "a"*'.match_in("aaa", ~input_prefix: #"x")
+)
+
+ If @rhombus(out) is provided as an output port for the
+ @rhombus(~unmatched_out) argument, the part of @rhombus(input) from its
+ beginning (including before @rhombus(start)) that precedes the match is
+ written to the port. All input up to @rhombus(end) is written to
+ @rhombus(out) if no match is found. This functionality is most useful
+ when @rhombus(input) is an input port.
+
+@examples(
+  ~eval: rx_eval
+  ~repl:
+    def out = Port.Output.open_string()
+    rx'"a"+'.match_in("before aaa after", ~unmatched_out: out)
+    out.get_string()
+)
+
+}
+
+@doc(
+  method (regexp :: RX).try_match(input :: Port.Input,
+                                  ~start: start :: Int = 0,
+                                  ~end: end :: maybe(Int) = #false,
+                                  ~input_prefix: input_prefix :: Bytes = #"",
+                                  ~unmatched_out: out :: maybe(Port.Output)
+                                                    = #false)
+    :: maybe(RXMatch)
+  method (regexp :: RX).try_match_in(....) :: maybe(RXMatch)
+){
+
+ Like @rhombus(RX.match) and @rhombus(RX.match_in), but no bytes are
+ consumed from @rhombus(input) if the pattern does not match.
+
+@examples(
+  ~eval: rx_eval
+  def p = Port.Input.open_string("hello")
+  rx'"hi"'.try_match(p)
+  p.peek_char()
+  rx'"hi"'.match(p)
+  p.peek_char()
+)
+
+}
+
+
+
+
+@doc(
+  method (regexp :: RX).matches(input :: String || Bytes || Port.Input,
+                                ~start: start :: Int = 0,
+                                ~end: end :: maybe(Int) = #false,
+                                ~input_prefix: input_prefix :: Bytes = #"")
+    :: List.of(String || Bytes)
+  method (regexp :: RX).split(input :: String || Bytes || Port.Input,
+                              ~start: start :: Int = 0,
+                              ~end: end :: maybe(Int) = #false,
+                              ~input_prefix: input_prefix :: Bytes = #"")
+    :: List.of(String || Bytes)
+){
+
+ Like @rhombus(RX.match_in), but finding all non-overlapping matches.
+ The @rhombus(RX.matches) method returns the found matches, and
+ @rhombus(RX.split) returns the complement, i.e., the strings that are
+ between matches. The result from @rhombus(RX.split) will start or end
+ with empty strings if the regexp matches the start or end of the input,
+ respectively.
+
+
+@examples(
+  ~eval: rx_eval
+  rx'any ["abc"] any'.matches("xbx ycy")
+  rx'any ["abc"] any'.matches(#"xbx ycy")
+)
+
+}
+
+
+@doc(
+  method (regexp :: RX).replace(
+    input :: String || Bytes,
+    insert :: (String || Bytes || Function.of_arity(1+num_captures)),
+    ~input_prefix: input_prefix :: Bytes = #""
+  ) :: String || Bytes
+  method (regexp :: RX).replace_all(....)
```

Comment:
I think some languages also allow users to provide multiple patterns and replacers in one query, since that's different than sequentially doing `replace_all` multiple times. Do we want to provide a similar operation?

Reply:
Sounds nice, if someone wants to implement that.